### ISSUE TITLE
"Large select" example vertically overflows in Forms component documentation

### ISSUE THREAD

**Issue by Chealer** [twbs/bootstrap#23769]
The second example in [the Sizing section of the documentation of the Forms component](https://getbootstrap.com/docs/4.0/components/forms/#sizing) starts with a dropdown which shows by default a "Large select" option. The bottom of this text is not visible in my Google Chrome 60, at default zoom level.

There is no problem with the 2 following selects.

**Comment by andresgalante** [comment]
Hi @Chealer, Can you send a screen capture please? I can't reproduce it

**Comment by houfio** [comment]
![screenshot from 2017-09-01 13-58-50](https://user-images.githubusercontent.com/26674115/29968827-b31b9c36-8f1d-11e7-90fe-57d55ac53390.png)

**Comment by andresgalante** [comment]
That's on windows, right?

**Comment by houfio** [comment]
Nope, I'm running Debian 9.

**Comment by Chealer** [comment]
Hi @andresgalante, here is a screenshot showing a worst case (each character is affected). 
![forms bootstrap large select - mozilla firefox](https://user-images.githubusercontent.com/486141/29971532-7e67cb76-8ef6-11e7-8bc6-dfd9ec4156c0.png)

This is taken from Firefox 55.0 on Windows 8 at default zoom level.

**Comment by andresgalante** [comment]
@Chealer @houfio I see! that really sucks! I'll start a virtual machine and try to fix it.

Thanks a lot for reporting it and sending the screenshots

@Johann-S I think this should be labled as "bug" and "css"

**Comment by Johann-S** [comment]
Yep my bad thanks @andresgalante 👍

**Comment by andresgalante** [comment]
@Chealer Are you sure you are running beta? 
This is where the height is set: https://github.com/twbs/bootstrap/blob/v4-dev/scss/_forms.scss#L172-L176

And on reboot the `line-height` is set to `inherit`

I can't reproduce the issue on [browserstack](www.browserstack.com) with any browser, sorry 😞 

IF you can guide me I can try send a PR, or you can actually send it yourself!

**Comment by Chealer** [comment]
@andresgalante, I am "running" whatever the page I linked to uses, which appears to be a beta indeed, though it is the default documentation the homepage links to.

I do not master CSS, it is hard for me to say why the display is wrong.

I even get overflow if I try with MSIE, though it is less important in this case (only some letters, as shown by houfio).

**Comment by andresgalante** [comment]
Sorry @Chealer if since I can't reproduce it I can't fix it. But this is a bug.

**Comment by Chealer** [comment]
No problem @andresgalante, thanks anyway for your diligent effort.

It should not be hard to reproduce though. I also see the issue with Chromium 60 on Debian 9.0. Only some letters are affected there (like houfio again).

**Comment by githubjeka** [comment]
**Windows 10, Chrome v60.0.3112.113 (64bit)
Display 1680x1050**

![image](https://user-images.githubusercontent.com/874234/29997234-f1d18c98-9016-11e7-8502-076349635a42.png)

**Edge Browser and Internet Explorer 
Large is same as Default**
![image](https://user-images.githubusercontent.com/874234/29997233-e1f248da-9016-11e7-9f05-cabb506adf43.png)

<hr>

To fix for me need recalucalculate height:

![image](https://user-images.githubusercontent.com/874234/29997285-1945aca4-9018-11e7-839c-2d6e6fb16375.png)

![image](https://user-images.githubusercontent.com/874234/29997286-23d9bcbe-9018-11e7-8294-4f535793ec19.png)

![image](https://user-images.githubusercontent.com/874234/29997339-393133fc-9019-11e7-85fd-dffba296e404.png)

instead of

```css
height: calc(2.3125rem + 2px);
```

**Comment by andresgalante** [comment]
@githubjeka Yeap, the bug is definitely on the height. 

If I can help you to create a PR and look for the root of this variable please let me know.

**Comment by Julieng50** [comment]
The bug is not only on select element, i get the same result with input text :/

**Comment by Borkason** [comment]
> @andresgalante : Sorry @Chealer if since I can't reproduce it I can't fix it. But this is a bug.

I can reproduce this with a fresh install of Windows 10 1703 and Egde. Do you have a 4K display or something? Maybe a DPI other than 100% makes this a non-reproducible error on your side?

**Comment by andresgalante** [comment]
I have a mac :)

**Comment by Borkason** [comment]
I suspected it. You said earlier that you created a VM. Could it be possible that the VM was running with a different DPI setting than 100% on that Mac? We have an iMac 5K at work and when you spin up new VMs in Parallels, the DPI setting is adjusted so everything is not only readable by ants. Could be that this is the issue why you can't reproduce this.

Sorry that I can't be of any other help, but debugging CSS stuff is really no my thing. I am glad that I can put Bootstrap to user without any modifications to CSS where I need it :)

**Comment by jrysig** [comment]
There is definitely problem with input and select controls. You can reproduce this problem with free virtual machines from Microsoft: https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/

Download Windows 7 with IE11 and Windows 10 with MS Edge images. Run it in VirtualBox, install latest Chrome and Firefox. Open Bootstrap 4 beta documentation https://getbootstrap.com/docs/4.0/components/forms/ and use Dev tools to change page content. Note that both images are running with 100% DPI (default) settings. You will receive:

Win 7, IE 11
![w7_ie11](https://user-images.githubusercontent.com/7038517/30733885-32a496ae-9f79-11e7-8c1d-5fc171528c47.png)

Win 7, FF 55
![w7_ff55](https://user-images.githubusercontent.com/7038517/30733893-3f329e8e-9f79-11e7-8df7-58c4a5493ddb.png)

Win 7, Chrome 61
![w7_chrome61_a](https://user-images.githubusercontent.com/7038517/30733916-4ee36b2e-9f79-11e7-8439-e6238d39ca02.png)

Win 7, Chrome 61
![w7_chrome61_b](https://user-images.githubusercontent.com/7038517/30733927-5ba5c4b0-9f79-11e7-8862-dd994a7d426d.png)

Win10, MS Edge
![w10_edge](https://user-images.githubusercontent.com/7038517/30734514-c3c0ade2-9f7b-11e7-94b4-b3b6995b865f.png)

**Comment by kjdion84** [comment]
This is still a big problem. Browserstack is not a reliable method for testing these things, especially when everyone who is using the most common browser on the most common OS is experiencing these issues even though you say you can't reproduce it.

**Comment by andresgalante** [comment]
Of course it is. it would be great if anyone sends a PR to fix it. I just can't reproduce it.

**Comment by kjdion84** [comment]
You can reproduce it on a Windows or Ubuntu VM.

**Comment by andresgalante** [comment]
I could, just not have the time to create a VM and make it run to fix this.

**Comment by kjdion84** [comment]
@Julieng50 solution works. However, it even works when only using:

```
select {
    padding-top: .4rem !important;
    padding-bottom: .4rem !important;
}
```

Modifying `input` fields is not necessary.

**Comment by wojtask9** [comment]
1. inputs
Issue occurs because `line-height` is smaller than input "carret". 
Someone can tell that `line-height` is 1.25 and `font size` is 1rem. But not for all fonts space needed to render those fonts is the same.

2. select
The main issue is `height` on "select" (select.font-control:not() ... selector)
It is almost correct for "Arial" but not necessary for others fonts.

how to fix this.
This is really tricky and for general purpose I don't know how. Using `height` for "inputs" causes vertical align issues. For "selects" the same situation.

@andresgalante 
Try different font-family and I believe that you can reproduce this issue

**Comment by wojtask9** [comment]
Found this https://iamvdo.me/en/blog/css-font-metrics-line-height-and-vertical-align
Maybe this help to understand issue more precisely and someone can make PR :)

**Comment by dracos** [comment]
I believe this bug was actually fixed in #23150 back in July, merged just *after* the beta was made. If you use bleeding edge v4-dev, the issue is not present. This might be the cause of some confusion of the above comments.

**Comment by Johann-S** [comment]
your're right @dracos already fixed with our last dist files from v4-dev see : https://codepen.io/Johann-S/pen/MEpBdR

**Comment by wojtask9** [comment]
select is fixed but this bug also mention on inputs

**Comment by dracos** [comment]
You could easily create a codepen and see that the height fix of #23150 applied to all `.form-control-lg` including inputs.

**Comment by wojtask9** [comment]
@dracos 
You are right. Sorry for noise.

**Comment by kjdion84** [comment]
Please stop closing issues that aren't fixed properly. The issue still exists for even normal selects in Chrome on Windows. Characters are still being cut off as the text is still not centered vertically properly.

Proof fiddle: https://codepen.io/anon/pen/WZpPPp
Proof image: 
![Imgur](https://i.imgur.com/JLH8mad.jpg)

**Comment by dracos** [comment]
The large select in your screenshot has nothing cut off, which is what this issue is about?
Nothing before in this issue has talked about the default select being cut off.

**Comment by Johann-S** [comment]
See my Codepen @dracos which use our last dist file : https://codepen.io/Johann-S/pen/MEpBdR

**Comment by dracos** [comment]
Yes I agree the text is cut off the bottom of the normal select, I can see that myself in IE. But this ticket and the screenshots above was about the `-lg` overhang, which was fixed by the PR I gave. It took me long enough to track that down that fix, and having git issues contain many overlapping different bugs does not help matters, witness the many confused tickets in this area.

**Comment by Johann-S** [comment]
yep that's right :+1: so close and I won't reopen that issue, track #23347 please

**Comment by kjdion84** [comment]
The `-lg` and normal selects should not be treated as separate issues. They conform to the same element, the only thing different is the sizing. Creating a solution that requires individual changes to each sub-style is not efficient in any way, ESPECIALLY considering we are now using `rem`. There should be a solution which, when applied to `select.form-control`, will scale efficiently no matter what sizing changes are made in regards to font/padding.

**Comment by dracos** [comment]
I agree. The issue of *this* issue, as it was described and followed up, was that `-lg` was using 2.3125rem rather than 2.875rem for its height, which was a plain bug due to it using the wrong variable, and that was fixed in https://github.com/twbs/bootstrap/pull/23150. #23347 seems like a more appropriate issue to track the general issue to do with sizing, *given* that the obvious issue with `-lg` has been fixed. I hope that explains thing properly, as nothing I've said is contradicting what you are saying.

**Comment by Chealer** [comment]
Regarding issue tracking, it is poor practice to use a report of several issues to track an important issue. Someone looking for a report of this bug will unlikely look at a report with a title as vague as "v4 form input issues".

On the other hand, I did not know the bug I reported here was caused by a bug in Bootstrap itself, so this ticket's title is not optimal to track the Bootstrap bug exposed.

I would recommend to either:

1. open a new ticket tracking the underlying bug
2. reopen this ticket and possibly retitle it
3. retitle #23347 with a more specific name
